client/wp-admin-scripts/onboarding-homepage-notice/index.js

```diff
@@ -0,0 +1,86 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { select, subscribe, dispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * WooCommerce dependencies
+ */
+import { getAdminLink } from '@woocommerce/navigation';
+
+/**
+ * Displays a notice on page save and updates the hompage options.
+ *
+ * Adapted from WSUWP Help Docs by Washington State University (GPL v2).
+ * https://github.com/washingtonstateuniversity/WSUWP-Plugin-Help-Docs/blob/master/src/_js/blocks/notices.js
+ */
+const onboardingHomepageNotice = () => {
+	const post = select( 'core/editor' ).getCurrentPost();
+
+	const noticeMeta = {
+		wasSavingPost: select( 'core/editor' ).isSavingPost(),
+		wasAutosavingPost: select( 'core/editor' ).isAutosavingPost(),
+		wasPublishingPost: select( 'core/editor' ).isPublishingPost(),
+		wasStatus: post.status,
+	};
+
+	subscribe( () => {
+		const isSavingPost = select( 'core/editor' ).isSavingPost();
+		const isAutosavingPost = select( 'core/editor' ).isAutosavingPost();
+		const isPublishingPost = select( 'core/editor' ).isPublishingPost();
+		const isStatus = select( 'core/editor' ).getCurrentPost().status;
+
+		// This triggers when the homepage is first published.
+		const shouldTriggerAdminNotice =
+			noticeMeta.wasSavingPost &&
```

Comment:
I'm not super familiar with all of these states, but it seems like there may be an opportunity to cut down this logic to:
```js
'publish' === isStatus && wasStatus !== isStatus
```

But if this causes issues, feel free to ignore this comment.

Reply:
Unfortunately, the status changes to `publish` before the post finishes saving, which is too early to clear the other message out.

However, I am able to simplify this down to:

`const shouldTriggerAdminNotice = 'publish' === isStatus && noticeMeta.wasPublishingPost && ! isStatus.isPublishingPost;`

which is much better.